recipe/patches/0003-link-protobuf-privately.patch

```diff
@@ -0,0 +1,45 @@
+From b334f2d910825d3518c853775e30ad332830b8ea Mon Sep 17 00:00:00 2001
+From: Mark Harfouche <mark.harfouche@gmail.com>
+Date: Sat, 9 Mar 2024 21:16:50 -0500
+Subject: [PATCH] link protobuf privately
+
+---
+ CMakeLists.txt | 10 +++++-----
+ 1 file changed, 5 insertions(+), 5 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index dc5cd52..1e39ed6 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -492,15 +492,15 @@ target_compile_definitions(onnx_proto PRIVATE ${ONNX_API_DEFINE})
+ 
+ if(ONNX_USE_LITE_PROTO)
+   if(TARGET protobuf::libprotobuf-lite)
+-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS})
++    target_link_libraries(onnx_proto PRIVATE protobuf::libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS})
```

Comment:
@h-vetinari do you understand why the error:

```
ld: illegal thread local variable reference to regular symbol __ZN6google8protobuf8internal15ThreadSafeArena13thread_cache_E for architecture x86_64
```
might appear on OSX after I added this patch linking with `PRIVATE` instead of `PUBLIC`

Reply:
First guess is that the symbol appears somewhere in the ABI of onnx (or even the API); plus making protobuf private hides the symbol from the linker so it fails when it's forced to search for it. 

In other words, if the protobuf usage were truly private to `onnx_proto`, this should work. Either it explicitly isn't private, or it's a bug in the privateness.

Reply:
Ah so maybe link onnx to proton if as well.